File: exercicios/03-native-module-comparativo/aluno-igorx8/ios/NativeDeviceIntegrity.swift
import Foundation
import React
import UIKit

@objc(NativeDeviceIntegrity)
final class NativeDeviceIntegrity: NSObject {
  @objc
  static func moduleName() -> String! {
    return "NativeDeviceIntegrity"
  }

  @objc
  func getDeviceName() -> String {
    return UIDevice.current.name
  }

  @objc
  func getOsVersion() -> String {
    return UIDevice.current.systemVersion
  }

  @objc
  func isRunningOnEmulator() -> Bool {
    #if targetEnvironment(simulator)
      return true
    #else
      return false
    #endif
  }

  @objc
  func getIntegritySnapshot(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    let snapshot: [String: Any] = [
      "platform": "ios",
      "isEmulator": isRunningOnEmulator(),
      "isDebuggable": _isDebugAssertConfiguration(),
      "deviceName": getDeviceName(),
      "osVersion": getOsVersion(),
    ]

    resolve(snapshot)
  }
}

File: exercicios/03-native-module-comparativo/aluno-igorx8/specs/NativeDeviceIntegrity.ts
import type { TurboModule } from 'react-native';
import { TurboModuleRegistry } from 'react-native';

export type DeviceIntegritySnapshot = {
  platform: string;
  isEmulator: boolean;
  isDebuggable: boolean;
  deviceName: string;
  osVersion: string;
};

export interface Spec extends TurboModule {
  getDeviceName(): string;
  getOsVersion(): string;
  isRunningOnEmulator(): boolean;
  getIntegritySnapshot(): Promise<DeviceIntegritySnapshot>;
}

export default TurboModuleRegistry.getEnforcing<Spec>('NativeDeviceIntegrity');
